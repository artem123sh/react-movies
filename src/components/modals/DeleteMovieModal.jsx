import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Modal from '../basic/Modal';
import Button from '../basic/Button';

const StyledActionContainer = styled.div`
  display: flex;
  justify-content: flex-end;
  margin-top: 3rem;
`;

const StyledMessage = styled.span`
  font-size: 1.5rem;
`;

const DeleteMovieModal = ({ onClose, deleteMovie }) => (
  <Modal title="delete movie" onClose={onClose}>
    <StyledMessage>Are you sure you want to delete this movie?</StyledMessage>
    <StyledActionContainer>
      <Button variant="contained" size="large" onClick={() => deleteMovie()}>Confirm</Button>
    </StyledActionContainer>
  </Modal>
);

DeleteMovieModal.propTypes = {
  onClose: PropTypes.func.isRequired,
  deleteMovie: PropTypes.func.isRequired,
};

export default DeleteMovieModal;
