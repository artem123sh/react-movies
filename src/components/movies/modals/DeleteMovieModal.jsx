import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Modal from '../../basic/Modal';
import Button from '../../basic/Button';

const StyledActionContainer = styled.div`
  display: flex;
  justify-content: flex-end;
  margin-top: 3rem;
`;

const StyledMessage = styled.span`
  font-size: 1.5rem;
`;

const StyledButton = styled(Button)`
  margin-left: 1rem;
`;

const DeleteMovieModal = ({ onClose }) => (
  <Modal title="delete movie" onClose={onClose}>
    <StyledMessage>Are you sure you want to delete this movie?</StyledMessage>
    <StyledActionContainer>
      <StyledButton variant="contained" size="large">Confirm</StyledButton>
    </StyledActionContainer>
  </Modal>
);

DeleteMovieModal.propTypes = {
  onClose: PropTypes.func.isRequired,
};

export default DeleteMovieModal;
