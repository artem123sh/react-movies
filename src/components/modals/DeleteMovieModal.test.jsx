import React from 'react';
import { render } from '@testing-library/react';
import DeleteMovieModal from './DeleteMovieModal';

describe('DeleteMovieModal', () => {
  test('renders DeleteMovieModal\'s snapshot', () => {
    const { asFragment } = render(
      <DeleteMovieModal onClose={() => {}} deleteMovie={() => {}} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
