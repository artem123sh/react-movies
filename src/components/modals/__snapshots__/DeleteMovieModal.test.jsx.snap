// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMovieModal renders DeleteMovieModal's snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ kYxXGN"
  />
  <div
    class="sc-AxiKw ghPDlE"
  >
    <div>
      <div
        class="sc-AxhUy fSLCtT"
      >
        <span
          class="sc-AxhCb fGYvSv"
        >
          â•³
        </span>
      </div>
      <div
        class="sc-AxheI iMsYLJ"
      >
        <span
          class="sc-AxjAm sc-AxgMl YYvbv"
        >
          delete movie
        </span>
        <div
          class="sc-Axmtr eqaOhC"
        >
          <span
            class="sc-fzoLsD jqWCJU"
          >
            Are you sure you want to delete this movie?
          </span>
          <div
            class="sc-fzozJi cpdHLS"
          >
            <button
              class="sc-AxmLO hfReIG"
              type="button"
            >
              Confirm
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
